## 1. Introduction

- This case study is based on FiveThirtyEight's article ["The Economic Guide To Picking A College Major" ](https://fivethirtyeight.com/features/the-economic-guide-to-picking-a-college-major/) published in 2014.
- For many American college students, enrolling in a degree often invovles taking on student debt, but a wrong choice of college degree could leave them worse off than if they have not gone to colleage at all.  A [2014 study by the New York Fed](https://libertystreeteconomics.newyorkfed.org/2014/09/the-value-of-a-college-degree.html#.VA6Uh_mwLwk) found that the lowest-earning 25 percent of college graduates earn less than about half of high school graduates — and the high school grads also had four years to make money while the college students were taking on student debt. 
- Therefore, we would like to find out if a quantified approach, based on empirical data, could help this demographic make a more informed decision on their choice of college major, from an ROI (return on investment) point of view.

##  2. The questions we are trying to answer

In this Github, we will attempt to answer the following questions:
- Which Majors would help the student earn the highest starting salary?
- Which Majors would offer the best employment prospects?
- Which Majors to avoid (lowest salary, worst employment prospects)?

## 3. Dataset

The dataset used in this case study comes from the American Community Survey on job outcome statistics based on college majors, stored as a database file.  The data spans a period of 2010-2012 for recent college grads only. Of note, the database file jobs.db contains only one table named recent_grads. 

Using  SQL (Structured Query Language), we can quickly retrieve the information need to answer the above questions such as which category of College Major would result in the highest starting salary, the prospects of employment etc. 



## 4. How this is done in SQL:

To do this, we will write SQL code to express our data request, then ask the SQLite DBMS software to run the code and display the results.



SELECT  Major_category, Major, Median, Unemployment_rate <br>
FROM recent_grads <br>
WHERE (Major_category = 'Engineering') OR (Major_category = 'Physical Sciences') <br>
ORDER BY Median ASC

In [1]:
import sqlite3
import pandas as pd
conn = sqlite3.connect("jobs.db")

q1 = "SELECT * FROM sqlite_master WHERE type='table';"
pd.read_sql_query(q1, conn)

type          name      tbl_name  rootpage  \
0  table  recent_grads  recent_grads         2   

                                                 sql  
0  CREATE TABLE "recent_grads" (\n"index" INTEGER...

In [2]:
q2 = "SELECT * FROM recent_grads LIMIT 5"
pd.read_sql_query(q2, conn)

index  Rank  Major_code                                      Major  \
0      0     1        2419                      PETROLEUM ENGINEERING   
1      1     2        2416             MINING AND MINERAL ENGINEERING   
2      2     3        2415                  METALLURGICAL ENGINEERING   
3      3     4        2417  NAVAL ARCHITECTURE AND MARINE ENGINEERING   
4      4     5        2405                       CHEMICAL ENGINEERING   

  Major_category  Total  Sample_size    Men  Women  ShareWomen      ...        \
0    Engineering   2339           36   2057    282    0.120564      ...         
1    Engineering    756            7    679     77    0.101852      ...         
2    Engineering    856            3    725    131    0.153037      ...         
3    Engineering   1258           16   1123    135    0.107313      ...         
4    Engineering  32260          289  21239  11021    0.341631      ...         

   Part_time  Full_time_year_round  Unemployed  Unemployment_rate  Median  \
0        270                  1207          37           0.018381  110000   
1        170                   388          85           0.117241   75000   
2        133                   340          16           0.024096   73000   
3        150                   692          40           0.050125   70000   
4       5180                 16697        1672           0.061098   65000   

   P25th   P75th  College_jobs  Non_college_jobs  Low_wage_jobs  
0  95000  125000          1534               364            193  
1  55000   90000           350               257             50  
2  50000  105000           456               176              0  
3  43000   80000           529               102              0  
4  50000   75000         18314              4440            972  

[5 rows x 22 columns]

## 5. Which Majors had the highest starting salary?

The following SQL code was used to generate the table below:

In [10]:
q2 = '''
SELECT  Major_category, Major, Median, Unemployment_rate FROM recent_grads ORDER BY Median DESC LIMIT 10
'''

pd.read_sql_query(q2, conn)

Major_category                                      Major  Median  \
0        Engineering                      PETROLEUM ENGINEERING  110000   
1        Engineering             MINING AND MINERAL ENGINEERING   75000   
2        Engineering                  METALLURGICAL ENGINEERING   73000   
3        Engineering  NAVAL ARCHITECTURE AND MARINE ENGINEERING   70000   
4        Engineering                       CHEMICAL ENGINEERING   65000   
5        Engineering                        NUCLEAR ENGINEERING   65000   
6           Business                          ACTUARIAL SCIENCE   62000   
7  Physical Sciences                 ASTRONOMY AND ASTROPHYSICS   62000   
8        Engineering                     MECHANICAL ENGINEERING   60000   
9        Engineering                     ELECTRICAL ENGINEERING   60000   

   Unemployment_rate  
0           0.018381  
1           0.117241  
2           0.024096  
3           0.050125  
4           0.061098  
5           0.177226  
6           0.095652  
7           0.021167  
8           0.057342  
9           0.059174

It is interesting to note that engineering degrees overwhelmingly topped the list in terms of salaries.
At the same time, some of the highest earning majors, such as Mining and Mineral Engineering had a rather high unemployment rate of 11.7%, reflecting the [decline of the mining industry](https://interestingengineering.com/top-mining-engineering-jobs-highest-salary) as the world shifts towards using renewable and sustainable energy.  From 2008-2012, [ 50,000 jobs were lost in the coal mining industry](https://www.washingtonpost.com/news/energy-environment/wp/2015/04/01/the-decline-in-coal-jobs-in-one-chart/?noredirect=on&utm_term=.e0c30835897b), and those skilled workers have either been finding new jobs within a related sector such as Natural Gas or have completely changed industries, such as Wind and Solar which added more than 79,000 jobs during the same period of time. <br>
Therefore it is worth noting that besides looking at the expected salary, students should also get a sense of the general trend in the industry in terms of growth, demand and job prospects.

## 6. Which Majors had the best employment prospects?

To answer this question, we run the following SQL code:<br>
<br>
SELECT  Major_category, Major, Median, Unemployment_rate<br>
FROM recent_grads<br>
WHERE Unemployment_rate < 0.03<br>
ORDER BY Unemployment_rate ASC

In [17]:
q3 = '''
SELECT  Major_category, Major, Median, 
ROUND((1 - Unemployment_rate)*100, 1) AS 'Employment Rate (%)'
FROM recent_grads 
WHERE Unemployment_rate < 0.10
ORDER BY Unemployment_rate ASC
LIMIT 20
'''

pd.read_sql_query(q3, conn)

Major_category  \
0               Computers & Mathematics   
1                Biology & Life Science   
2       Agriculture & Natural Resources   
3                             Education   
4                           Engineering   
5                   Law & Public Policy   
6                             Education   
7                           Engineering   
8       Agriculture & Natural Resources   
9                     Physical Sciences   
10                    Physical Sciences   
11                          Engineering   
12                          Engineering   
13                    Physical Sciences   
14                          Engineering   
15                          Engineering   
16  Industrial Arts & Consumer Services   
17             Psychology & Social Work   
18                          Engineering   
19               Biology & Life Science   

                                                Major  Median  \
0                    MATHEMATICS AND COMPUTER SCIENCE   42000   
1                                              BOTANY   37000   
2                                        SOIL SCIENCE   35000   
3          EDUCATIONAL ADMINISTRATION AND SUPERVISION   34000   
4           ENGINEERING MECHANICS PHYSICS AND SCIENCE   58000   
5                                     COURT REPORTING   54000   
6                       MATHEMATICS TEACHER EDUCATION   34000   
7                               PETROLEUM ENGINEERING  110000   
8                                 GENERAL AGRICULTURE   40000   
9                          ASTRONOMY AND ASTROPHYSICS   62000   
10               ATMOSPHERIC SCIENCES AND METEOROLOGY   35000   
11                                  MATERIALS SCIENCE   60000   
12                          METALLURGICAL ENGINEERING   73000   
13                                        GEOSCIENCES   36000   
14        MATERIALS ENGINEERING AND MATERIALS SCIENCE   52000   
15                 INDUSTRIAL PRODUCTION TECHNOLOGIES   46000   
16  ELECTRICAL, MECHANICAL, AND PRECISION TECHNOLO...   38400   
17                                  SOCIAL PSYCHOLOGY   36000   
18              ENGINEERING AND INDUSTRIAL MANAGEMENT   44000   
19                                           GENETICS   40000   

    Employment Rate (%)  
0                 100.0  
1                 100.0  
2                 100.0  
3                 100.0  
4                  99.4  
5                  98.8  
6                  98.4  
7                  98.2  
8                  98.0  
9                  97.9  
10                 97.8  
11                 97.7  
12                 97.6  
13                 97.6  
14                 97.2  
15                 97.2  
16                 97.1  
17                 97.0  
18                 96.6  
19                 96.6

When we look at the majors with the lowest unemployment rate, no clear trend can be seen, as it spans a variety of majors.
If we rank Employment Rate by Major_category, we find that the top three major categories are Law & Public Policy, Engineering and Physical Sciences.

In [33]:
q4 = '''
SELECT  Major_category, 
Total,
ROUND((1 - Unemployment_rate)*100, 1) AS 'Employment Rate (%)'
FROM recent_grads 
GROUP BY Major_category
ORDER BY Unemployment_rate ASC
'''

pd.read_sql_query(q4, conn)

Major_category   Total  Employment Rate (%)
0                   Law & Public Policy    1148                 98.8
1                           Engineering    2339                 98.2
2                     Physical Sciences    1792                 97.9
3                                Health  209394                 95.5
4             Humanities & Liberal Arts    3079                 95.3
5   Industrial Arts & Consumer Services   18498                 94.0
6               Computers & Mathematics  128319                 93.7
7                     Interdisciplinary   12296                 92.9
8           Communications & Journalism  213996                 92.5
9                Biology & Life Science    1762                 91.4
10                                 Arts    3340                 91.1
11                             Business    3777                 90.4
12      Agriculture & Natural Resources    4361                 90.3
13                       Social Science  139247                 90.1
14                            Education     818                 89.2
15             Psychology & Social Work    3014                 89.1

The SQL code to obtain this output is:<br>
<br>
SELECT  Major_category, Major, Median, Unemployment_rate<br>
FROM recent_grads<br>
WHERE Unemployment_rate < 0.03<br>
ORDER BY Median DESC

## 7. Which Majors have the lowest salary and worst employment prospects?

First we find rank majors by unemployment rate:

In [38]:
q5 = '''
SELECT  Major_category, 
Major, 
Median, 
ROUND(Unemployment_rate*100, 1) as 'Unemployment Rate %'
FROM recent_grads
ORDER BY Unemployment_rate DESC
LIMIT 10
'''

pd.read_sql_query(q5, conn)

Major_category                                       Major  \
0               Engineering                         NUCLEAR ENGINEERING   
1       Law & Public Policy                       PUBLIC ADMINISTRATION   
2   Computers & Mathematics  COMPUTER NETWORKING AND TELECOMMUNICATIONS   
3  Psychology & Social Work                         CLINICAL PSYCHOLOGY   
4       Law & Public Policy                               PUBLIC POLICY   
5   Computers & Mathematics                  COMMUNICATION TECHNOLOGIES   
6               Engineering              MINING AND MINERAL ENGINEERING   
7   Computers & Mathematics    COMPUTER PROGRAMMING AND DATA PROCESSING   
8            Social Science                                   GEOGRAPHY   
9               Engineering                                ARCHITECTURE   

   Median  Unemployment Rate %  
0   65000                 17.7  
1   36000                 15.9  
2   36400                 15.2  
3   25000                 14.9  
4   50000                 12.8  
5   35000                 12.0  
6   75000                 11.7  
7   41300                 11.4  
8   38000                 11.3  
9   40000                 11.3

While the categories of majors here are a mixed bag, we note that they tend to be rather niche, which could imply limited employment opportunities.

Next, we run an SQL query to find the majors with the lowest salaries:

In [40]:
q6 = '''
SELECT  Major_category, 
Major, 
Median, 
ROUND(Unemployment_rate*100, 1) as 'Unemployment Rate %'
FROM recent_grads
ORDER BY Median ASC
LIMIT 10
'''

pd.read_sql_query(q6, conn)

Major_category                                          Major  \
0                  Education                                LIBRARY SCIENCE   
1   Psychology & Social Work                          COUNSELING PSYCHOLOGY   
2   Psychology & Social Work                         EDUCATIONAL PSYCHOLOGY   
3   Psychology & Social Work                            CLINICAL PSYCHOLOGY   
4     Biology & Life Science                                        ZOOLOGY   
5                       Arts                         DRAMA AND THEATER ARTS   
6  Humanities & Liberal Arts                       COMPOSITION AND RHETORIC   
7  Humanities & Liberal Arts                        OTHER FOREIGN LANGUAGES   
8  Humanities & Liberal Arts                    ANTHROPOLOGY AND ARCHEOLOGY   
9                     Health  COMMUNICATION DISORDERS SCIENCES AND SERVICES   

   Median  Unemployment Rate %  
0   22000                 10.5  
1   23400                  5.4  
2   25000                  6.5  
3   25000                 14.9  
4   26000                  4.6  
5   27000                  7.8  
6   27000                  8.2  
7   27500                 10.7  
8   28000                 10.3  
9   28000                  4.8

It can be seen here that majors in Psychology and Social Work as well as Humanities & Liberal Arts tend to command the lowest salaries.

## 8. Refining the search, step by step

Certain majors may have lower admission requirements, resulting in larger student enrolments.  Here we will find out the full time and part time jobs these majors result in, as well as whether they result in college or non-college level paying jobs.

In [81]:
q7 = '''
SELECT  Major_category, 
Major, 
Total,
Full_time,
ROUND((CAST(Full_time as float)/CAST(Total as float)) * 100, 1) as Full_time_ratio,
Part_time
FROM recent_grads
ORDER BY Full_time_ratio DESC
LIMIT 30
'''

pd.read_sql_query(q7, conn)

Major_category  \
0               Computers & Mathematics   
1                             Education   
2   Industrial Arts & Consumer Services   
3                           Engineering   
4                           Engineering   
5                           Engineering   
6   Industrial Arts & Consumer Services   
7   Industrial Arts & Consumer Services   
8                              Business   
9                             Education   
10                             Business   
11                          Engineering   
12                    Physical Sciences   
13                          Engineering   
14                          Engineering   
15                          Engineering   
16                            Education   
17                             Business   
18                          Engineering   
19                          Engineering   
20      Agriculture & Natural Resources   
21      Agriculture & Natural Resources   
22              Computers & Mathematics   
23      Agriculture & Natural Resources   
24                             Business   
25  Industrial Arts & Consumer Services   
26              Computers & Mathematics   
27                             Business   
28              Computers & Mathematics   
29                             Business   

                                                Major   Total  Full_time  \
0                    MATHEMATICS AND COMPUTER SCIENCE     609        584   
1          EDUCATIONAL ADMINISTRATION AND SUPERVISION     804        733   
2                               MILITARY TECHNOLOGIES     124        111   
3         MECHANICAL ENGINEERING RELATED TECHNOLOGIES    4790       4175   
4                  INDUSTRIAL PRODUCTION TECHNOLOGIES    4631       3988   
5           NAVAL ARCHITECTURE AND MARINE ENGINEERING    1258       1069   
6                               CONSTRUCTION SERVICES   18498      15690   
7   ELECTRICAL, MECHANICAL, AND PRECISION TECHNOLO...    2435       2057   
8                 OPERATIONS LOGISTICS AND E-COMMERCE   11732       9639   
9                  TEACHER EDUCATION: MULTIPLE LEVELS   14443      11734   
10      MANAGEMENT INFORMATION SYSTEMS AND STATISTICS   18713      15141   
11                          ARCHITECTURAL ENGINEERING    2825       2277   
12                                       OCEANOGRAPHY    2418       1931   
13             MISCELLANEOUS ENGINEERING TECHNOLOGIES    8804       7001   
14                                NUCLEAR ENGINEERING    2573       2038   
15                              PETROLEUM ENGINEERING    2339       1849   
16                      MATHEMATICS TEACHER EDUCATION   14237      11259   
17                                            FINANCE  174506     137921   
18           INDUSTRIAL AND MANUFACTURING ENGINEERING   18968      14879   
19                             MECHANICAL ENGINEERING   91227      71298   
20                         PLANT SCIENCE AND AGRONOMY    7416       5798   
21              AGRICULTURE PRODUCTION AND MANAGEMENT   14240      11119   
22    COMPUTER ADMINISTRATION MANAGEMENT AND SECURITY    8066       6289   
23                       NATURAL RESOURCES MANAGEMENT   13773      10722   
24                                  ACTUARIAL SCIENCE    3777       2924   
25           TRANSPORTATION SCIENCES AND TECHNOLOGIES   15150      11688   
26           COMPUTER PROGRAMMING AND DATA PROCESSING    4168       3204   
27                                         ACCOUNTING  198633     151967   
28                               INFORMATION SCIENCES   11913       9105   
29                   MARKETING AND MARKETING RESEARCH  205211     156668   

    Full_time_ratio  Part_time  
0              95.9          0  
1              91.2          0  
2              89.5          0  
3              87.2        247  
4              86.1        597  
5              85.0        150  
6              84.8       1751  
7              84.5        287  
8              82.2       1183  
9   

While majors such as MATHEMATICS AND COMPUTER SCIENCE and EDUCATIONAL ADMINISTRATION AND SUPERVISION have more than 90% of their graudates secure full time jobs, the enrolment for these majors are relatively small.  We can adjust for this by restricting the search list to majors with larger enrolment numbers, which could suggest easier admission requirements.  The top 30 majors had 60,000 or more students, and we can use the WHERE function to 
implement this filter criterion:

In [79]:
q7 = '''
SELECT  Major_category, 
Major, 
Total,
Full_time,
ROUND((CAST(Full_time as float)/CAST(Total as float)) * 100, 1) as Full_time_ratio
FROM recent_grads
WHERE Total > 60000
ORDER BY Full_time_ratio DESC
'''

pd.read_sql_query(q7, conn)

Major_category  \
0                              Business   
1                           Engineering   
2                              Business   
3                              Business   
4                              Business   
5                              Business   
6                                Health   
7                             Education   
8                   Law & Public Policy   
9               Computers & Mathematics   
10          Communications & Journalism   
11                       Social Science   
12          Communications & Journalism   
13                            Education   
14                          Engineering   
15                          Engineering   
16                                 Arts   
17                       Social Science   
18              Computers & Mathematics   
19                       Social Science   
20  Industrial Arts & Consumer Services   
21                    Physical Sciences   
22            Humanities & Liberal Arts   
23            Humanities & Liberal Arts   
24                    Physical Sciences   
25             Psychology & Social Work   
26            Humanities & Liberal Arts   
27                                 Arts   
28               Biology & Life Science   
29                                 Arts   

                                            Major   Total  Full_time  \
0                                         FINANCE  174506     137921   
1                          MECHANICAL ENGINEERING   91227      71298   
2                                      ACCOUNTING  198633     151967   
3                MARKETING AND MARKETING RESEARCH  205211     156668   
4          BUSINESS MANAGEMENT AND ADMINISTRATION  329927     251540   
5                                GENERAL BUSINESS  234590     171385   
6                                         NURSING  209394     151191   
7                            ELEMENTARY EDUCATION  170862     123177   
8            CRIMINAL JUSTICE AND FIRE PROTECTION  152824     109970   
9                                COMPUTER SCIENCE  128319      91485   
10                                     JOURNALISM   72619      51411   
11                                      ECONOMICS  139247      96567   
12                                 COMMUNICATIONS  213996     147335   
13                              GENERAL EDUCATION  143718      98408   
14                         ELECTRICAL ENGINEERING   81527      55450   
15                            GENERAL ENGINEERING   61152      41235   
16              COMMERCIAL ART AND GRAPHIC DESIGN  103480      67448   
17               POLITICAL SCIENCE AND GOVERNMENT  182621     117709   
18                                    MATHEMATICS   72397      46399   
19                                      SOCIOLOGY  115433      73475   
20  PHYSICAL FITNESS PARKS RECREATION AND LEISURE  125074      77428   
21          MULTI-DISCIPLINARY OR GENERAL SCIENCE   62052      37850   
22                                   LIBERAL ARTS   71369      43401   
23                                        HISTORY  141951      84681   
24                                      CHEMISTRY   66530      39509   
25                                     PSYCHOLOGY  393735     233205   
26                ENGLISH LANGUAGE AND LITERATURE  194673     114386   
27                                      FINE ARTS   74440      42764   
28                                        BIOLOGY  280709     144512   
29                                          MUSIC   60633      29010   

    Full_time_ratio  
0              79.0  
1              78.2  
2              76.5  
3              76.3  
4              76.2  
5              73.1  
6              72.2  
7              72.1  
8              72.0  
9              71.3  
10             70.8  
11             69.3  
12             68.8  
13             68.5  
14             68.0  
15             67.4  
16             65.2  
17             64.5  
18             64.1  
19             63.

However, the concern here is that some of these majors may result in low-paying jobs.  We need not worry however, as this search can be further refined by restricting the search results only to jobs that command above-average salaries.  This is done by inserting a subquery for Median > (SELECT AVG(Median) from recent_grads) as part of the WHERE statement:

In [68]:
q7 = '''
SELECT  Major_category, 
Major, 
Total,
Full_time,
ROUND((CAST(Full_time as float)/CAST(Total as float)) * 100, 1) as Full_time_ratio,
Median
FROM recent_grads
WHERE Total > 60000 AND Median > (SELECT AVG(Median) from recent_grads)
ORDER BY Full_time_ratio DESC
'''

pd.read_sql_query(q7, conn)

Major_category                   Major   Total  Full_time  \
0                 Business                 FINANCE  174506     137921   
1              Engineering  MECHANICAL ENGINEERING   91227      71298   
2                 Business              ACCOUNTING  198633     151967   
3                   Health                 NURSING  209394     151191   
4  Computers & Mathematics        COMPUTER SCIENCE  128319      91485   
5           Social Science               ECONOMICS  139247      96567   
6              Engineering  ELECTRICAL ENGINEERING   81527      55450   
7              Engineering     GENERAL ENGINEERING   61152      41235   
8  Computers & Mathematics             MATHEMATICS   72397      46399   

   Full_time_ratio  Median  
0             79.0   47000  
1             78.2   60000  
2             76.5   45000  
3             72.2   48000  
4             71.3   53000  
5             69.3   47000  
6             68.0   60000  
7             67.4   56000  
8             64.1   45000

The series of incremental filters has helped us narrow down the list from the initial 173 majors to 9 majors, which we believe to offer a combination of reasonable admission requirements, have a strong ratio of graduates with full-time jobs, and offer prospects of an above-average salary.

As a hygiene check, we run a query to cross-check the average salary of the dataset, which works out to $40,151.

In [77]:
q7 = '''
SELECT AVG(Median) FROM recent_grads
'''

pd.read_sql_query(q7, conn)

AVG(Median)
0  40151.445087

## Conclusion

In this Github, we have demonstrated a number of SQL queries and techniques to rank college majors in terms of best and worst salaries and employment prospects.  We then ran a series of incrementally refined searches, culminating in a list of 9 college majors which we believe offer the most attractive return on investment in terms of ease of admission, likelihood of securing full-time employment and commanding an above-average salary.

With an adequate dataset, SQL presents a flexible and quantifiable tool to answer a variety of questions - in this case, empowering future college students to make more informed choices of which college major to pursue.